homeassistant/components/withings/common.py

```diff
@@ -724,15 +813,16 @@ async def async_set_data(self, attribute: WithingsAttribute, value: Any) -> None
         else:
             state = value
 
-        self._hass.states.async_set(
-            entity_id=entity_id,
-            new_state=state,
+        self._state_data[entity_id] = StateData(
+            state=state,
             attributes={
                 "friendly_name": attribute.friendly_name,
                 "unit_of_measurement": attribute.unit_of_measurement,
             },
         )
 
+        await async_update_entity(self._hass, entity_id)
```

Comment:
Don't do this, this is for users to use. I suggest you use the dispatcher helper. Let entities subscribe to it inside async_added_to_hass:

```python
    @callback
    def receive_data(self, new_state):
        """Receive new data.""""
        self._state = new_state
        self.async_write_ha_state()

    async def async_added_to_hass(self):
        """Register update dispatcher."""
        self.async_on_remove(
            async_dispatcher_connect(self.hass, SIGNAL_UPDATE_DATA, self.receive_data)
        )
```